### PR TITLE
Document new experimental ingestion streaming APIs

### PR DESCRIPTION
**Pull request by reta**
### Description
Document new experimental ingestion streaming APIs

### Issues Resolved
Closes https://github.com/opensearch-project/opensearch-api-specification/issues/537

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: spec/namespaces/_core.yaml [hunk 1]
```diff
@@ -76,6 +76,57 @@ paths:
       responses:
         '200':
           $ref: '#/components/responses/bulk@200'
+  /_bulk/stream:
+    post:
+      operationId: bulk.stream.0
+      x-operation-group: bulk
```
Comment (nhtruong) [inline spec/namespaces/_core.yaml:82]: The running linter will throw an error on the `operationId` and the `x-operation-group`.
I think you want to do `bulk_stream` for `x-operation-group` and `bulk_stream.0` for `operationId`. Then on the generated client, there will be a `client.bulk_stream()` method, which is different from `client.bulk()`

**Comment by nhtruong** [comment]
Please add tests to these operations as well.

File: spec/namespaces/_core.yaml (hunk 1 above)
Comment (reta) [reply spec/namespaces/_core.yaml:82]: Thanks a lot for the hints, @nhtruong !

File: spec/namespaces/_core.yaml [hunk 2]
```diff
@@ -76,6 +76,57 @@ paths:
       responses:
         '200':
           $ref: '#/components/responses/bulk@200'
+  /_bulk/stream:
+    post:
+      operationId: bulk_stream.0
+      x-operation-group: bulk_stream
+      x-version-added: '2.17'
+      description: Allows to perform multiple index/update/delete operations using request / response streaming
```
Comment (dblock) [inline spec/namespaces/_core.yaml:84]:
```suggestion
      description: Allows to perform multiple index/update/delete operations using request/response streaming.
```

File: spec/namespaces/_core.yaml [hunk 3]
```diff
@@ -76,6 +76,57 @@ paths:
...
+      requestBody:
+        $ref: '#/components/requestBodies/bulk'
+      responses:
+        '200':
+          $ref: '#/components/responses/bulk@200'
+    put:
+      operationId: bulk_stream.1
+      x-operation-group: bulk_stream
+      x-version-added: '2.17'
+      description: Allows to perform multiple index/update/delete operations using request / response streaming
```
Comment (dblock) [inline spec/namespaces/_core.yaml:109]:
```suggestion
      description: Allows to perform multiple index/update/delete operations using request/response streaming.
```

File: spec/namespaces/_core.yaml [hunk 4]
```diff
@@ -3217,6 +3268,23 @@ components:
...
       x-default: '1'
+    bulk::query.batch_size:
+      in: query
+      name: batch_size
+      description: |-
+        Specifies how many bulk operations should be accumulated into a batch before sending the batch to data nodes.
+      schema:
+        $ref: '../schemas/_common.yaml#/components/schemas/BatchSize'
+      style: form
+      x-default: '1'
```
Comment (dblock) [inline spec/namespaces/_core.yaml:3279]:
```suggestion
      x-default: 1
```

**Review by dblock — CHANGES_REQUESTED** [review]
Good start. Needs tests, please!

File: tests/default/_core/bulk_stream.yaml [hunk 5]
```diff
@@ -0,0 +1,54 @@
+$schema: ../../../json_schemas/test_story.schema.yaml
```
Comment (dblock) [inline tests/default/_core/bulk_stream.yaml:1]: File should be `_core/bulk/stream.yaml`.

File: tests/default/indices/bulk_stream.yaml [hunk 6]
```diff
@@ -0,0 +1,60 @@
+$schema: ../../../json_schemas/test_story.schema.yaml
```
Comment (dblock) [inline tests/default/indices/bulk_stream.yaml:1]: Same, `bulk/stream.yaml`.

**Review by dblock — CHANGES_REQUESTED** [review]
You'll need to update the 2.17 version used to the released one, and switch the next version to 2.18.

**Comment by reta** [comment]
> So does it actually stream data? 🤔

It does, I am not sure how we could express that in OpenAPI

**Comment by dblock** [comment]
> > So does it actually stream data? 🤔
> 
> It does, I am not sure how we could express that in OpenAPI

I don't think we need to, but do clients automatically do it? AFAIK we don't have code in our tooling that does anything special.

**Comment by reta** [comment]
```
ERROR   _core/bulk/stream.yaml (tests/default/_core/bulk/stream.yaml)
    ERROR   CHAPTERS
        ERROR   Create an index.
            PASSED  REQUEST BODY
            ERROR   RESPONSE STATUS (Expected status 200, but received 500: application/json. The engine does not support HTTP streaming, unable to serve uri [/_bulk/stream] and method [POST])
            SKIPPED RESPONSE PAYLOAD BODY
            SKIPPED RESPONSE PAYLOAD SCHEMA
            SKIPPED RESPONSE OUTPUT VALUES
        SKIPPED Delete document in an index.
        SKIPPED Bulk document CRUD.
```

@dblock need your advice here: streaming (as of today) require dedicated transport plugin. Here are 2 options we could make it work:
 - ignore the test if particular plugin is not installed, or
 - install (and make default) the streaming transport

**Comment by reta** [comment]
> I don't think we need to, but do clients automatically do it? AFAIK we don't have code in our tooling that does anything special.

Oh, we don't need to actually, no special treatment required

**Comment by dblock** [comment]
> * install (and make default) the streaming transport

I think this is the right answer. The tooling in this repo should support streaming just like we support various content types.

File: tests/default/indices/bulk/stream.yaml [hunk 7]
```diff
@@ -0,0 +1,60 @@
+$schema: ../../../../json_schemas/test_story.schema.yaml
```
Comment (dblock) [inline tests/default/indices/bulk/stream.yaml:1]: You'll have to move tests in the plugins folder as well from tests/default.

**Review by dblock — APPROVED** [review]

**Pull request merged**